lib/auth/cognito_stub_client.rb

```diff
@@ -41,21 +46,47 @@ def self.stub_client
     Aws::CognitoIdentityProvider::Client.new(stub_responses: true)
   end
 
+  def self.switch_to_cognito
+    # Exit if we are already a cognito client
+    return "already cognito" if false?(SelfService.service(:cognito_stub))
+    # Exit if there isn't a cognito client available to switch to
+    return "no cognito client" unless SelfService.service_present?(:real_client)
+
+    real_client = SelfService.service(:real_client)
+    SelfService.register_service(name: :cognito_client, client: real_client)
+    SelfService.register_service(name: :cognito_stub, client: 'false')
+    "switched to cognito"
+  end
+
+  def self.switch_to_stub
+    # Exit if we are in production we don't want a stub in production
+    return "in prod" if Rails.env.production?
+    # Exit if we're already a stub client
+    return "already stub" unless false?(SelfService.service(:cognito_stub))
+
+    real_client = SelfService.service(:cognito_client)
+    SelfService.register_service(name: :real_client, client: real_client)
+    SelfService.register_service(name: :cognito_client, client: stub_client)
+    setup_stubs
+    SelfService.register_service(name: :cognito_stub, client: 'true')
+    "switched to stub"
```

Comment:
These return strings aren't used for `switch_to_stub` and `switch_to_cognito`. The flash messages are set independently above